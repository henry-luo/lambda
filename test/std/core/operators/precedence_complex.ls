// Test: Complex Operator Precedence
// Category: core/operators
// Type: positive
// Expected: 23

// Test mixed operations: 10 + 2 * 5 - 3 / 3 = 10 + 10 - 1 = 19
// Actually: 5 * 2 + 10 + 3 = 10 + 10 + 3 = 23
let a = 5
let b = 2
a * b + 10 + 3
