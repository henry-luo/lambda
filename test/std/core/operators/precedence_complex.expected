09:15:47 [NOTICE] Start transpiling test/std/core/operators/precedence_complex.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr +.3:int]
      [binary expr +.3:int]
        [binary expr *.5:int]
          [primary expr:int,const:1]
            [ident:a:int,const:1]
          [primary expr:int,const:1]
            [ident:b:int,const:1]
        [primary expr:int,const:1]
          [const: int]
      [primary expr:int,const:1]
        [const: int]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/precedence_complex.ls' to MIR
C2MIR init end           -- 136 usec
test/std/core/operators/precedence_complex.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2458, parse tokens -- 1490
  C2MIR preprocessor end    -- 707 usec
  C2MIR parser end          -- 837 usec
  C2MIR context checker end -- 928 usec
  C2MIR generator end       -- 985 usec
C2MIR compiler end                -- 1022 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/precedence_complex.ls' executed: #####
23
